src/components/ProductsList/ProductsList.tsx

```diff
@@ -3,35 +3,96 @@ import React, { useEffect, useState } from 'react';
 import PrevIcon from '@assets/icons/Chevron_Left.svg?react';
 import NextIcon from '@assets/icons/Chevron_Right.svg?react';
 
-import { useFilterContext } from '@/hooks/useFilterContext';
-import { useProductsDataContext } from '@/hooks/useProductsDataContext';
+import { useFetch } from '@/hooks/useFetch';
+import { useFiltersContext } from '@/hooks/useFiltersContext';
+import type { Product } from '@/interfaces/Product';
+import type { FetchProductsInterface } from '@/utils/http';
+import { fetchProducts } from '@/utils/http';
 
-import { Pagination } from '../Pagination/Pagination';
+import { ProductCard } from '../ProductCard/ProductCard';
 import { SearchBar } from '../SearchBar/SearchBar';
 
 import styles from './ProductsList.module.css';
 
+const productsInitialValue: FetchProductsInterface = { products: [], total: 0 };
+
 const productCardsOnPage = 8;
+const isMobile = window.innerWidth < 1290;
 
 function ProductsList() {
-    const { filteredAndSortedProductsData } = useFilterContext();
-    const { isErrorFetchingProductsData, isLoadingProductsData } = useProductsDataContext();
     const [currentPage, setCurrentPage] = useState(1);
+    const { search, sortBy, category } = useFiltersContext();
+    const { fetchData, fetchedData, isFetching, error } = useFetch(productsInitialValue, fetchProducts);
+    const qtyOfPages = Math.ceil((fetchedData ? fetchedData.total : 0) / productCardsOnPage);
+
+    const [data, setData] = useState<Product[]>([]);
 
-    const isVisiblePagesButtons = !isErrorFetchingProductsData && !isLoadingProductsData && filteredAndSortedProductsData.length > 0;
+    // test
+
+    useEffect(() => {
+        if (isMobile) {
+            if (currentPage === 1) {
+                if (fetchedData) {
+                    setData(fetchedData?.products);
+                }
+            } else {
+                if (fetchedData) {
+                    setData((previous) => [...previous, ...fetchedData.products]);
+                }
+            }
+        } else {
+            if (fetchedData) {
+                setData(fetchedData?.products);
+            }
+        }
+
+        return () => {};
+        // if remove 'currentPage' everything work fine bun i don't know how to do it
+    }, [fetchedData, currentPage]);
```

Comment:
why do u7 need to remove currentPage from deps?

Reply:
Infinite scroll and filters don't work correctly